boards/BMS/Src/App/App_Accumulator.c

```diff
@@ -270,9 +309,38 @@ void App_Accumulator_RunOnTick100Hz(struct Accumulator *const accumulator)
             // Calculate min/max/segment voltages
             App_Accumulator_CalculateVoltageStats(accumulator);
 
+            // Find cells to discharge
+            App_Accumulator_CalculateCellsToDischarge(accumulator);
+
             // Write to configuration register to configure cell discharging
-            accumulator->write_cfg_registers();
-            accumulator->disable_discharge();
+            accumulator->write_cfg_registers(accumulator->cells_to_discharge);
+
+            if (accumulator->mute_discharge)
+            {
+                // Disable cell discharging
+                accumulator->disable_discharge();
+                accumulator->discharge_pwm_ticks += 1;
+
+                if (accumulator->discharge_pwm_ticks >= DISCHARGE_TICKS_OFF)
+                {
+                    // Cell discharging disabled duty cycle portion is finished
+                    accumulator->mute_discharge      = false;
+                    accumulator->discharge_pwm_ticks = 0;
+                }
+            }
+            else
+            {
+                // Enable cell discharging
+                accumulator->enable_discharge();
+                accumulator->discharge_pwm_ticks += 1;
+
+                if (accumulator->discharge_pwm_ticks >= DISCHARGE_TICKS_ON)
+                {
+                    // Cell discharging enabled duty cycle portion is finished
+                    accumulator->mute_discharge      = true;
+                    accumulator->discharge_pwm_ticks = 0;
+                }
+            }
```

Comment:
i'd put this whole block in its own function to keep the higher level state machine functionality clear to the reader

nit - if possible i'd move the cell balancing algorithm to its own file so a reader can see exactly how the system works

Reply:
yeah this was definitely a bit slapped together, good call ill do some refactoring